packages/frontend/src/hooks/usePositions.ts

```diff
@@ -112,12 +134,33 @@ export const usePositions = () => {
     [isWethToken0, swaps],
   )
 
+  const mintedDebt = useMemo(() => {
+    return oSqueethBal?.isGreaterThan(0) && positionType === PositionType.LONG
+      ? oSqueethBal.minus(squeethAmount)
+      : oSqueethBal
+  }, [oSqueethBal.toString(), positionType, squeethAmount.toString()])
+
+  const shortDebt = useMemo(() => {
+    return positionType === PositionType.SHORT ? squeethAmount : new BigNumber(0)
+  }, [positionType, squeethAmount.toString()])
+
+  const longSqthBal = useMemo(() => {
+    return mintedDebt.gt(0) ? oSqueethBal.minus(mintedDebt) : oSqueethBal
+  }, [positionType, oSqueethBal.toString(), mintedDebt.toString()])
+
+  const lpDebt = useMemo(() => {
+    return depositedSqueeth.minus(withdrawnSqueeth).isGreaterThan(0)
+      ? depositedSqueeth.minus(withdrawnSqueeth)
+      : new BigNumber(0)
+  }, [positionType, depositedSqueeth.toString(), withdrawnSqueeth.toString()])
+
   const { finalSqueeth, finalWeth } = useMemo(() => {
-    const finalSqueeth = squeethAmount.minus(depositedSqueeth).plus(withdrawnSqueeth)
+    // dont include LPed amount will be the correct short amount
+    const finalSqueeth = squeethAmount
```

Comment:
wdym correct short amount? If you are just looking for how much someone is short that should not include both minted and LPed

Reply:
But the squeethAmount is already the short amount that someone has, the LPed and minted amount was not accounted as squeethAmount in the beginning so we dont need to subtract LPed and minted amount here

Reply:
Ok, comment was confusing because it only called out LP not minting as well

Reply:
lol gotcha, will edit this